components/table/ReportTable.tsx

```diff
@@ -359,15 +361,20 @@ function ReportTable({
     timesheets,
 }: ReportTableProps): JSX.Element {
     const { user } = useContext(UserContext)
+    const { role } = useContext(AuthContext)
 
     const firstDay = dateTimeToShortISODate(DateTime.now().startOf("month"))
     const lastDay = dateTimeToShortISODate(DateTime.now().endOf("month"))
 
     const [startDate, setStartDate] = useState<string>(firstDay)
     const [endDate, setEndDate] = useState<string>(lastDay)
 
-    const reportsResponse = useTimesheetEntries(user, startDate, endDate)
-    const allEntries = reportsResponse.isSuccess ? reportsResponse.data : []
+    const reportsResponse =
+        role === Role.ADMIN
+            ? useTimesheetEntries(user, startDate, endDate)
+            : false
+    const allEntries =
+        reportsResponse && reportsResponse.isSuccess ? reportsResponse.data : []
```

Comment:
Type of `reportsResponse` is a little confusing here. Maybe it could be undefined if the role isn't admin? that's easy to verify in the next ternary too.